test_ros2cli/test/ros2service_test_configuration.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+
+import os
+import re
+import sys
+
+from launch_testing_ros.tools import basic_output_filter
+
+sys.path.append(os.path.dirname(__file__))
+
+from cli_test_configuration import CLITestConfiguration  # noqa
+from fixture_actions import get_add_two_ints_server_action  # noqa
+
+
+def get_add_two_ints_call_output(*, a=0, b=0):
+    return [
+        'requester: making request: ' +
+        'example_interfaces.srv.AddTwoInts_Request(a={}, b={})'.format(a, b),
+        '',
+        'response:',
+        'example_interfaces.srv.AddTwoInts_Response(sum={})'.format(a + b),
+        ''
+    ]
+
+
+def get_test_configurations(rmw_implementation):
+    return [
+        CLITestConfiguration(
+            command='service',
+            arguments=['list'],
+            fixture_actions=[get_add_two_ints_server_action(),
+                             get_add_two_ints_server_action(service_name='_add_two_ints')],
+            expected_output=itertools.chain(
+                # Cope with launch internal ROS 2 node.
+                itertools.repeat(re.compile(r'/launch_ros/.*parameter.*'), 6),
```

Comment:
We are checking for an exact number of lines here (6), of something that may easily change.
I don't know if it's better checking just if some lines are in the output, or checking for exact output.
This apply in many places, but I don't have an strong opinion about what's better.

Check lines in output pro: Updating the test is needed less often
Check lines in output con: Maybe, the test should have been updated (adding extra checks) after a change in other place, and it wasn't.

Reply:
> Check lines in output con: Maybe, the test should have been updated (adding extra checks) after a change in other place, and it wasn't.

 That's the main reason why I went for a restrictive rather than a permissive test. If we trust CLI release testing on these tests, I'm more comfortable being as precise as possible regarding how they should behave.